CMP-2582: Update CO install documentation to include ROSA support (#76938)
openshift/openshift-docs: modules/compliance-operator-rosa-installation.adoc

// Module included in the following assemblies:
//
// * security/compliance_operator/co-management/compliance-operator-installation.adoc

:_mod-docs-content-type: PROCEDURE
[id="installing-compliance-operator-rosa_{context}"]
= Installing the Compliance Operator on ROSA hosted control planes (HCP)

> As of the Compliance Operator 1.5.0 release, the operator is tested against Red Hat
> OpenShift Service on AWS (ROSA) using hosted control plane topologies (HCP).

Reviewer: * Capitalize "Operator" per [repo term glossary](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc#operator).
* There's a [common-attribute](https://github.com/openshift/openshift-docs/blob/main/_attributes/common-attributes.adoc?plain=1#L164) available for the full ROSA name.
```suggestion
As of the Compliance Operator 1.5.0 release, the Operator is tested against {product-rosa} using hosted control plane topologies (HCP).
```
There are also some for HCP, if you like:
```
// Hosted control planes related attributes
:hcp-capital: Hosted control planes
:hcp: hosted control planes
```

ROSA HCP clusters have restricted access to the control plane, which is managed
by Red Hat. By default, the Compliance Operator will schedule to nodes within

> the `master` node pool, which isn't available in ROSA HCP installations. This

Reviewer: Avoid contractions per [SSG](https://redhat-documentation.github.io/supplementary-style-guide/#contractions):
```suggestion
the `master` node pool, which is not available in ROSA HCP installations. This
```

requires you to configure the `Subscription` object in a way that allows the

> operator to schedule on available node pools. This step is necessary for a

Reviewer: ```suggestion
Operator to schedule on available node pools. This step is necessary for a
```

successful installation on ROSA HCP clusters.

.Prerequisites

* You must have `admin` privileges.

.Procedure

. Define a `Namespace` object:
+

> .Example `namespace-object.yaml`

Reviewer: ```suggestion
.Example `namespace-object.yaml` file
```
I won't comment on all of these, but apply consistently like this for any similar block titles.

[source,yaml]
----
apiVersion: v1
kind: Namespace
metadata:
  labels:
    openshift.io/cluster-monitoring: "true"
    pod-security.kubernetes.io/enforce: privileged <1>
  name: openshift-compliance
----

> <1> in {product-title} {product-version}, the pod security label must be set to `privileged` at the namespace level.

Reviewer: ```suggestion
<1> In {product-title} {product-version}, the pod security label must be set to `privileged` at the namespace level.
```

. Create the `Namespace` object:
+
[source,terminal]
----
$ oc create -f namespace-object.yaml
----

. Define an `OperatorGroup` object:
+
.Example `operator-group-object.yaml`
[source,yaml]
----
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: compliance-operator
  namespace: openshift-compliance
spec:
  targetNamespaces:
  - openshift-compliance
----

. Create the `OperatorGroup` object:
+
[source,terminal]
----
$ oc create -f operator-group-object.yaml
----

. Define a `Subscription` object:
+
.Example `subscription-object.yaml`
[source,yaml]
----
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: compliance-operator-sub
  namespace: openshift-compliance
spec:
  channel: "stable"
  installPlanApproval: Automatic
  name: compliance-operator
  source: redhat-operators
  sourceNamespace: openshift-marketplace
  config:
    nodeSelector:
      node-role.kubernetes.io/worker: "" <1>
----

> <1> update the operator deployment to deploy on `worker` nodes.

Reviewer: ```suggestion
<1> Update the Operator deployment to deploy on `worker` nodes.
```

. Create the `Subscription` object:
+
[source,terminal]
----
$ oc create -f subscription-object.yaml
----

.Verification

> . Verify the installation succeeded by inspecting the CSV file:

Reviewer: ```suggestion
. Verify the installation succeeded by inspecting the cluster service version (CSV) object:
```

// Module included in the following assemblies:
//
// * security/compliance_operator/co-management/compliance-operator-installation.adoc

:_mod-docs-content-type: PROCEDURE
[id="installing-compliance-operator-cli_{context}"]
= Installing the Compliance Operator on ROSA

As of the Compliance Operator 1.5.0 release, the operator is tested against Red Hat
OpenShift Service on AWS (ROSA) using hosted control plane topologies (HCP).

> ROSA HCP clusters restrict access to the control plane, which is managed by Red

Reviewer 2: Maybe you want to say
`ROSA HCP clusters **have** restrict access to the control plane, which is managed by Red Hat`
